<?xml version="1.0"?>
<launch>
    <arg name="ns" value="R_001"/>
    <arg name="prefix_ns" value="R_001/dsr"/>

    <!--<node name="rviz" pkg="rviz" type="rviz" args="-d $(find snu_doosan)/rviz/snu_doosan.rviz"/>-->

<!-- 
##############################################################################################################
    SNU-ASMR 데모1 실행: Cube(ar_marker_1 on top) Pick and Place (Single AR Marker)
        - launch 1. snu_doosan/launch/realsense_setup.launch.xml
        - launch 2. snu_doosan/launch/ar_tracker_single.launch.xml
        - node 1.   snu_doosan/snu_drl_interface.py
        - node 2.   snu_doosan/snu_object_tracker.py // (*) offset unit -> [m], [deg]
##############################################################################################################
-->
    <include file="$(find snu_doosan)/launch/realsense_setup.launch.xml" ns="$(arg ns)"/>
    <include file="$(find snu_doosan)/launch/ar_tracker_single.launch.xml" ns="$(arg ns)"/>

    <!--<node pkg="snu_doosan" type="snu_drl_interface.py" name="test" ns="$(arg ns)" output='screen'/>-->

    <node pkg="snu_doosan" type="snu_object_tracker.py" name="snu_object_tracker" ns="$(arg ns)" output='screen'>
        <param name="reference_frame"       value="base_0"/>
        <param name="object_frame"          value="ar_marker_0"/>
        <param name="target_frame"          value="ar_target_0"/>
        <param name="end_effector_frame"    value="link6"/>
        <param name="offset_from_target/x"  value="0.0"/>
        <param name="offset_from_target/y"  value="0.0"/>
        <param name="offset_from_target/z"  value="0.20"/>
        <param name="offset_from_target/rx" value="180.0"/>
        <param name="offset_from_target/ry" value="0.0"/>
        <param name="offset_from_target/rz" value="90.0"/>
    </node>
    
    <!--<node pkg="snu_doosan" type="drl_gripping.py" name="drl_gripping" ns="$(arg ns)" output="screen"/>-->


<!-- 
##############################################################################################################
    SNU-ASMR 데모2 실행: Grasping an Object on the 3DP Printing-bed (Bundle AR Markers)
        - launch 1. snu_doosan/launch/realsense_setup.launch.xml
        - launch 2. snu_doosan/launch/ar_tracker_bundle.launch.xml
        - node 1.   snu_doosan/snu_object_tracker.py // (*) offset unit -> [m], [deg]
##############################################################################################################  
-->
    <!--<include file="$(find snu_doosan)/launch/realsense_setup.launch.xml" ns="$(arg ns)"/>
    <include file="$(find snu_doosan)/launch/ar_tracker_bundles.launch.xml" ns="$(arg ns)"/>

    <node pkg="snu_doosan" type="snu_object_tracker.py" name="snu_object_tracker" ns="$(arg ns)" output='screen'>
        <param name="reference_frame"       value="base_0"/>
        <param name="object_frame"          value="ar_marker_0"/>
        <param name="target_frame"          value="ar_target_0"/>
        <param name="offset_from_target/x"  value="0.0"/>
        <param name="offset_from_target/y"  value="-0.175"/>
        <param name="offset_from_target/z"  value="0.055"/>
        <param name="offset_from_target/rx" value="180.0"/>
        <param name="offset_from_target/ry" value="-90.0"/>
        <param name="offset_from_target/rz" value="90.0"/>
    </node>-->



<!-- 
##############################################################################################################
    Unity <-> ROS Test: Communication(Pub&Sub), Control(Unity -> ROS)
        - node 1. snu_doosan/unity_test.py
##############################################################################################################  
-->
    <!--<node pkg="snu_doosan" type="unity_test.py" name="unity_test" ns="$(arg ns)" output='screen'/>-->



<!-- 
##############################################################################################################
    Testing a Single Node:
##############################################################################################################  
-->
    <!--<node name="Pub_client" pkg="rospy_websocker_client" type="ROSClient_Publisher.py" ns='R_001' output="screen"/>-->
    <!--<node name="Sub_client" pkg="rospy_websocker_client" type="ROSClient_Subscriber.py" ns='R_001'     output="screen"/>-->

    <!--<node pkg="snu_doosan" type="snu_dsr_joy.py" name="joy_controller" ns="$(arg ns)" output='screen'/>-->
    
    <!--<include file="$(find snu_doosan)/launch/realsense_setup.launch.xml" ns="$(arg ns)"/>-->
    <!--<include file="$(find rtabmap_ros)/launch/rgbd_mapping.launch" ns="$(arg prefix_ns)"/>-->

    <!--<include file="$(find snu_doosan)/launch/ar_tracker_single.launch.xml" ns="$(arg ns)"/>-->
    
    <!--<node pkg="snu_doosan" type="snu_moveit_commander.py" name="snu_moveit_commander" ns="$(arg ns)" output='screen'/>-->

    <!--<include file="$(find snu_doosan)/launch/dsr_moveit.launch" ns="$(arg ns)"/>
    <include file="$(find rospy_websocker_client)/launch/Publisher.launch" ns="$(arg ns)"/>
    <include file="$(find rospy_websocker_client)/launch/Subscriber.launch" ns="$(arg ns)"/>
    <node pkg="snu_doosan" type="snu_moveit_commander.py" name="snu_moveit_commander" ns="$(arg ns)"/>
    <node pkg="snu_doosan" type="snu_dsr_digital_io_controller.py" name="snu_dsr_digital_io_controller" ns="$(arg ns)"/>
    <node pkg="snu_doosan" type="snu_dsr_joy.py" name="snu_dsr_joy" ns="$(arg ns)"/>-->


    
</launch>

